src/Uno.Extensions.Validation/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Uno.Extensions.Validation;
+
+public static class ServiceCollectionExtensions
+{
+	public static IServiceCollection AddInstanceTypeInfo<TEntity>(
+	this IServiceCollection services
+	)where TEntity : class
+	{
+		return services
+			.AddSingleton<IInstanceType>(sp => new IInstanceTypeWrapper<TEntity>(sp));
+		
+	}
+}
+
+internal interface IInstanceType : IValidator
+{
+	Type InstanceType { get; }
+}
+
+internal record IInstanceTypeWrapper<T>(IServiceProvider Services) : IInstanceType
```

Comment:
This isn't an interface, so change the name to ValidatorTypedInstance (and do same for serialization)